pkg/apis/monitoring/v1/thanos_types.go

```diff
@@ -38,7 +38,9 @@ const (
 // +kubebuilder:printcolumn:name="Paused",type="boolean",JSONPath=".status.paused",description="Whether the resource reconciliation is paused or not",priority=1
 // +kubebuilder:subresource:status
 
-// ThanosRuler defines a ThanosRuler deployment.
+// The `ThanosRuler` custom resource definition (CRD) defines a desired [Thanos Ruler](https://github.com/thanos-io/thanos/blob/main/docs/components/rule.md) setup to run in a Kubernetes cluster.
+//
+// A `ThanosRuler` instance requires at least one compatible Prometheus API endpoint (either Thanos Querier or Prometheus services).
```

Comment:
For consistency with Prometheus/Alertmanager, we can also mention the PrometheusRule selection for thanos ruler

```
// The resource defines via label and namespace selectors which `PrometheusRule` objects should be associated with the deployed Thanos Ruler instances.
```

Reply:
Simon and Nicolas might have some suggestions, so let's wait for them and then I will add this together with it.

Reply:
Agree with @ArthurSens. Something along these lines?

```
The resource defines via label and namespace selectors 
which `PrometheusRule` objects should be associated to the 
deployed Thanos Ruler instances.
````